packages/commonwealth/client/scripts/views/pages/Community/Groups/common/GroupForm/GroupForm.tsx

```diff
@@ -322,39 +469,101 @@ const GroupForm = ({
     setRequirementSubForms([...allRequirements]);
   };
 
-  const validateSubForms = () => {
+  const validateSubForms = async () => {
     const updatedSubForms = [...requirementSubForms];
 
-    requirementSubForms.map((subForm, index) => {
-      try {
-        // HACK ALERT: this type of validation change should be done internally by zod, by we are doing this
-        // manually using javascript
-        const schema = getRequirementSubFormSchema(
-          subForm.values.requirementType,
-        );
-        if (subForm.values.requirementType === '') {
-          schema.pick({ requirementType: true }).parse(subForm.values);
-        } else {
-          schema.parse(subForm.values);
-        }
+    await Promise.all(
+      requirementSubForms.map(async (subForm, index) => {
+        try {
+          // HACK ALERT: this type of validation change should be done internally by zod, by we are doing this
+          // manually using javascript
+          const schema = getRequirementSubFormSchema(
+            subForm.values.requirementType,
+          );
+          if (subForm.values.requirementType === '') {
+            schema.pick({ requirementType: true }).parse(subForm.values);
+          } else {
+            schema.parse(subForm.values);
+          }
 
-        updatedSubForms[index] = {
-          ...updatedSubForms[index],
-          errors: {},
-        };
-      } catch (e: any) {
-        const zodError = e as ZodError;
-        const errors = {};
-        zodError.errors.map((x) => {
-          errors[x.path[0]] = x.message;
-        });
-
-        updatedSubForms[index] = {
-          ...updatedSubForms[index],
-          errors: errors as any,
-        };
-      }
-    });
+          updatedSubForms[index] = {
+            ...updatedSubForms[index],
+            errors: {},
+          };
+
+          const key = Object.keys(subForm.values)[3];
+          const isCosmos =
+            subForm.values.requirementType === CW_SPECIFICATIONS.CW_721;
+
+          const contract_address = subForm.values.requirementContractAddress;
+          const amount = subForm.values.requirementAmount;
+          const evmId = parseInt(subForm.values.requirementChain); // requirement.data.source.evm_chain_id;
+          const cosmosId = subForm.values.requirementChain; // requirement.data.source.cosmos_chain_id;
+          const node = await axios.get(
```

Comment:
The query parameter here doesn't do anything I think -> this seems like it just fetches all chain nodes for every requirement which is inefficient. We should use either `app.config.nodes` if it is already loaded or make a single ReactQuery call to fetch these nodes. @masvelio is it fine to use `app.config.nodes` or is this something we are trying to deprecate?

Reply:
at some point we will be willing to move to react-query but at this moment we are using app.config.nodes in multiple places so it is ok to use it. When the time will come, we will be refactoring all those occurences